lib/Parser/OglParser.php

```diff
@@ -23,32 +23,34 @@
 class OglParser
 {
     private $hints;
+    private $matrix;
 
-    public function __construct($hints) {
+    public function __construct($hints, $matrix) {
         $this->hints = $hints;
+        $this->matrix = $matrix;
     }
 
-    public function parse($filename) {
+    public function parse($filename, $commit = null) {
         $handle = fopen($filename, "r");
```

Comment:
In that case we call `parse($filename, null)`, because we can set `null` doesn't mean that we have to put it as a default parameter.

My concern is that if we want to add a new, mandatory parameter to `parse`, we'll have false-positives problems because of this default parameter. No default parameter == no false-positive for future improvements.